forms/NumericField.php

```diff
@@ -2,13 +2,63 @@
 
 /**
  * Text input field with validation for numeric values. Supports validating
- * the numeric value as to the {@link i18n::get_locale()} value.
+ * the numeric value as to the {@link i18n::get_locale()} value, or an
+ * overridden locale specific to this field.
  * 
  * @package forms
  * @subpackage fields-formattedinput
  */
 class NumericField extends TextField {
 
+	/**
+	 * @var string
+	 */
+	protected $locale = null;
+
+	public function setValue($value, $data = array()) {
+		require_once "Zend/Locale/Format.php";
+
+		// If passing in a non-string number, or a value
+		// directly from a dataobject then localise this number
+		if ((is_numeric($value) && !is_string($value)) || 
+			($value && $data instanceof DataObject)
+		){
+			$locale = new Zend_Locale($this->getLocale());
+			$this->value = Zend_Locale_Format::toNumber($value, array('locale' => $locale));
+		} else {
+			// If an invalid number, store it anyway, but validate() will fail
+			$this->value = $this->clean($value);
+		}
+		return $this;
+	}
+
+	/**
+	 * In some cases and locales, validation expects non-breaking spaces
+	 *
+	 * @param string $input
+	 * @return string The input value, with all spaces replaced with non-breaking spaces
+	 */
+	public static function clean($input) {
```

Comment:
> It's also used inconsistently as static (in tests) and instance method (in IsNumeric).

Apparently it's fine (and not at all bad practice) to call static methods non-statically, only vice versa. We do this a lot with `Object::config`. :) It's like the negative margins of the backend dev world.